python/ray/data/datasource/file_meta_provider.py

```diff
@@ -284,6 +284,7 @@ def _get_block_metadata(
         if (
             prefetched_metadata is not None
             and len(prefetched_metadata) == num_fragments
+            and all(m is not None for m in prefetched_metadata)
```

Comment:
This is needed for Parquet datasource, as previously it depends on Parquet datasource to just return a empty list if metadata is unknown.